ddl/job_table.go

```diff
@@ -194,28 +165,41 @@ func hasSysDB(job *model.Job) bool {
 	return false
 }
 
-func (d *ddl) handleUpgradingState(se *sess.Session, job *model.Job) error {
-	if !d.stateSyncer.IsUpgradingState() {
-		if !job.IsPausedBySystem() || hasSysDB(job) {
-			return nil
+func (d *ddl) processJobDuringUpgrade(sess *sess.Session, job *model.Job) (err error, isRunnable bool) {
+	if d.stateSyncer.IsUpgradingState() {
+		if hasSysDB(job) {
+			return err, false
```

Comment:
I think we need to handle this job which is means we can return true for `isRunnable`.
Besides, I think we'd better return `nil` instead of `err`.

Reply:
good one

Reply:
Actually, it also depends how we design the behavior. We could say that all DDL jobs should not be running during upgrade.